### ISSUE TITLE
[SUPPORT] nested group access to parent model and validity using expressionProperties

### ISSUE THREAD

**Issue by jogelin** [formly-js/ngx-formly#758]
**I'm submitting a ...**  (check one with "x")
```
[ ] bug report => search github for a similar issue or PR before submitting
[X] feature request
[X] support request
```

**Current behavior**
NA

**Expected behavior**
Based on a form description like this: 
```
[
    {
      key: 'reference',
      type: 'input',
      templateOptions: {
        label: 'Reference',
        required: true
      }
    },
    {
      key: 'persons',
      type: 'person-search-field',
      // TODO: here the model is an array, how can I access to the reference value
      expressionProperties: {
        'templateOptions.reference': 'model.reference',
        'templateOptions.referenceIsValid': 'form.reference.valid'
      },
      fieldArray: {
        templateOptions: {
          label: 'Search Person'
        }
      }
    } 
  ]
```

I would like to:
* provide the value of the reference field to the type `person-search-field` (this value is needed to search a person)
* disable the type `person-search-field` if the field `reference`  is not valid (really like formControl.valid)

I based my implementation of `person-search-field` on the `repeat-field` type but I am using a separate form to search for a person and then, I add it to the `formArray`. Is it a good way of doing that ?

**Minimal reproduction of the problem with instructions**
You can find the example on [StackBlitz](https://stackblitz.com/edit/angular-formly-template-rc-32?file=app%2Fapp.component.ts)

Thx in advance, this lib is very useful. 

By the way, I am beginner using formly and I think it is very useful to have a lot of examples. Are you open for PR that add example in your doc ? For example "Nested Searching Type"...

**Comment by aitboudad** [comment]
> Are you open for PR that add example in your doc ? 

PR welcomed :smile:

**Comment by jogelin** [comment]
Good ;), I'll try to add more specifics example when I solve a scenario.

But first, I need to solve this one ;)

For the moment, I did an ugly thing in my custom type like:

```
this.form
      .get(this.to.referenceKey)
      .statusChanges.pipe(distinctUntilChanged())
      .subscribe(status => {
        if (status === 'VALID') {
          this.enable();
        } else {
          this.disable();
        }
        this.cdRef.detectChanges();
      });
```

But this create a string link between this type and my form structure which I don't like...

**Comment by aitboudad** [comment]
Do you know about `formState` ?

**Comment by aitboudad** [comment]
anyway here is an easy way to solve you issue:
```ts
     expressionProperties: {
        'templateOptions.reference': () => this.model.reference,
        'templateOptions.disabled': () => {
          return this.form.get('reference').valid;
        },
     },
```

you can also use `formState` which is kind of a context you can use among fields:
```ts
  model: any = {reference:null};
  options: FormlyFormOptions = {
    formState: {
       formModel: this.model
    }
  };
  ...
  expressionProperties: {
    'templateOptions.reference': (model, formState) => formState.formModel.reference,
    'templateOptions.disabled': (model, formState) => formState.formModel.isRefValid,
  },
```

**Comment by jogelin** [comment]
I was playing a bit with it too but without success. I love this library because I feel that it can cover
 a lot of cases/scenario but the main difficulties is HOW to do it and WHAT is the best way...it is why I think you have to support a lot.

I think your approach of the documentation, by providing a list of examples, is really the best way to learn !

Thx again, I will test your solution !

**Comment by aitboudad** [comment]
see https://stackblitz.com/edit/angular-formly-template-rc-32-k1nda6

**Comment by aitboudad** [comment]
managing repeating fields has been simplified see https://formly-js.github.io/ngx-formly/examples/advanced/repeating-section all you have to do is extend `FieldArrayType`.

**Comment by aitboudad** [comment]
one more thing, if you want the seach field to not be part of your form, you can define it without key and set `formControl` property:
```ts
{
  type: 'input',
  formControl: new FormControl(),
  templateOptions: {
    label: 'Person Name',
    required: true
  },
},
```

**Comment by IustiNN** [comment]
@aitboudad Hello, I'm trying to use repeat section and i have this error... 
/node_modules/@ngx-formly/core/core"' has no exported member 'FieldArrayType'.

**Comment by jogelin** [comment]
As usual, your proposition works very well. I updated the [StackBlitz](https://stackblitz.com/edit/angular-formly-template-rc-32?file=app%2Fperson-search-field.component.ts)

However, I have an issue in my app....it almost the same as the stackblitz example but I don't know why, the `'templateOptions.disabled': () => this.to.disabled` (on my input on my separate form contained by my custom type) is not re-evaluate (but I could see that the value of `this.to.disabled` is changing). 

At the beginning I thought it was because formly only re-evaluate expression within a form. It means that if I modified `reference` of the main form, the expressionProperties of the separate form containing the field `person name` is not called......but it works on stackblitz.....:s

Any ideas about that before I close this issue ?
